modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Tags/MongoTagRepository.cs

```diff
@@ -56,9 +57,24 @@ public MongoTagRepository(IMongoDbContextProvider<ICmsKitMongoDbContext> dbConte
                cancellationToken: cancellationToken);
         }
 
-        public Task<List<Tag>> GetAllRelatedTagsAsync([NotNull] string entityType, [NotNull] string entityId, Guid? tenantId = null, CancellationToken cancellationToken = default)
+        public virtual async Task<List<Volo.CmsKit.Tags.Tag>> GetAllRelatedTagsAsync(
+            [NotNull] string entityType,
+            [NotNull] string entityId,
+            Guid? tenantId = null,
+            CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            var entityTagIds = await DbContext.EntityTags.AsQueryable()
+                .Where(q => q.EntityId == entityId)
```

Comment:
Wouldn't be better filtering tenantId in first query?
```suggestion
                .Where(q => q.EntityId == entityId && q.TenantId == tenantId)
```